qiskit/circuit/quantumcircuit.py

```diff
@@ -1895,33 +1878,52 @@ def _unroll_param_dict(self, value_dict):
                 unrolled_value_dict.update(zip(param, value))
         return unrolled_value_dict
 
-    def _bind_parameter(self, parameter, value):
-        """Assigns a parameter value to matching instructions in-place."""
-        for (instr, param_index) in self._parameter_table[parameter]:
-            instr.params[param_index] = instr.params[param_index].bind({parameter: value})
+    def _assign_parameter(self, parameter, value):
```

Comment:
This reduces the number of LOC but doesn't change the behavior, it's a simple refactor. 
<img width="879" alt="Screen Shot 2020-09-28 at 5 58 43 PM" src="https://user-images.githubusercontent.com/10730254/94490511-62e75180-01b4-11eb-9c71-702c43c84407.png">

Reply:
`assign_parameters` binds or assigns depending on the type of the input

Reply:
I think maybe we may need to agree on what "bind" and "assign" means and then make sure our methods follow those meanings. I think the one I suggested is used in mathematics and computer science. Unless you want to make the method distinction in this pr it may make sense to keep with the status quo.